addons/account_check_printing/models/account_payment.py

```diff
@@ -258,51 +258,74 @@ def prepare_vals(invoice, partials):
                 invoice_sign = -1
                 partial_field = 'credit_amount_currency'
 
-            if invoice.currency_id.is_zero(invoice.amount_residual):
+            amount_residual = max(invoice.amount_residual - current_amount, 0) if current_amount else invoice.amount_residual
```

Comment:
Why do you need the max, or to check the current amount?
```suggestion
            amount_residual = invoice.amount_residual - current_amount
```

Reply:
Indeed. It's useless. It was used in the first version of the fix to prevent having a negative amount_residual when the amount of the check was higher than the amount of the bills.

The `min` already makes sure that it won't happen. :sweat_smile: 
```python
current_amount = min(remaining, inv_to_check.currency_id._convert(
    from_amount=inv_to_check.amount_residual,
    to_currency=self.currency_id,
))
```